# Hypothesis Testing for Bombay Hospitality Ltd.

## Background
Bombay Hospitality Ltd. operates a franchise model for producing exotic Norwegian dinners throughout New England. The operating cost for a franchise in a week (W) is given by the equation:
\[
W = 1000 + 5X
\]
where \( X \) represents the number of units produced in a week. Recent feedback from restaurant owners suggests that this cost model may no longer be accurate, as their observed weekly operating costs are higher.

## Objective
To investigate the restaurant owners' claim about the increase in weekly operating costs using hypothesis testing.

## Data Provided
- **Theoretical weekly operating cost model:** \( W = 1000 + 5X \)
- **Sample of 25 restaurants** with a mean weekly cost of Rs. 3050
- **Number of units produced in a week (X)** follows a normal distribution with a mean (\( \mu \)) of 600 units and a standard deviation (\( \sigma \)) of 25 units

## Assignment Tasks

### 1. State the Hypotheses Statement
- **Null Hypothesis (H0):** The mean weekly operating cost is equal to the theoretical mean weekly cost based on the cost model.
  \[
  H_0: \mu = 1000 + 5 \cdot 600 = 4000
  \]
- **Alternative Hypothesis (H1):** The mean weekly operating cost is higher than the theoretical mean weekly cost.
  \[
  H_1: \mu > 4000
  \]

### 2. Calculate the Test Statistic
We will calculate the test statistic (t) using the formula:
\[
t = \frac{\bar{x} - \mu}{\frac{\sigma}{\sqrt{n}}}
\]
where:
- \( \bar{x} = 3050 \) (sample mean weekly cost)
- \( \mu = 4000 \) (theoretical mean weekly cost)
- \( \sigma = 5 \cdot 25 = 125 \) (standard deviation of the weekly cost based on units produced)
- \( n = 25 \) (sample size)

### 3. Determine the Critical Value
For a one-tailed test at a significance level of 5% (\( \alpha = 0.05 \)), the critical value from the standard normal (Z) distribution table is approximately 1.645.

### 4. Make a Decision
Compare the calculated test statistic with the critical value to decide whether to reject the null hypothesis.

### 5. Conclusion
Based on the comparison, conclude whether there is strong evidence to support the restaurant owners' claim that the weekly operating costs are higher than the model suggests.

---

Below is the Python code that performs these steps and provides the result of the hypothesis test.


In [1]:
import numpy as np
import scipy.stats as stats

# Data provided
sample_mean = 3050
theoretical_mean = 1000 + 5 * 600  # W = $1,000 + $5X for X = 600 units
sigma = 5 * 25  # Standard deviation
n = 25
alpha = 0.05

# Step 2: Calculate the Test Statistic
test_statistic = (sample_mean - theoretical_mean) / (sigma / np.sqrt(n))

# Step 3: Determine the Critical Value
critical_value = stats.norm.ppf(1 - alpha)  # Z critical value for one-tailed test

# Step 4: Make a Decision
print(f"Test Statistic: {test_statistic:.4f}")
print(f"Critical Value: {critical_value:.4f}")

if test_statistic > critical_value:
    conclusion = "Reject the null hypothesis. There is strong evidence to suggest that the weekly operating costs are higher than the model suggests."
else:
    conclusion = "Fail to reject the null hypothesis. There is not enough evidence to suggest that the weekly operating costs are higher than the model suggests."

print(conclusion) 


Test Statistic: -38.0000
Critical Value: 1.6449
Fail to reject the null hypothesis. There is not enough evidence to suggest that the weekly operating costs are higher than the model suggests.


## Hypothesis Testing Results

### Test Statistic and Critical Value

- **Test Statistic:** -38.0000
- **Critical Value:** 1.6449

### Decision

Since the test statistic (-38.0000) is less than the critical value (1.6449), we **fail to reject the null hypothesis**.

### Conclusion

There is not enough evidence to suggest that the weekly operating costs are higher than the cost model suggests. The test statistic being significantly lower than the critical value indicates that the observed mean cost is not significantly higher than the theoretical cost at the 5% significance level.

---

This result suggests that the current cost model is still a valid representation of the weekly operating costs for the restaurants, based on the provided sample data.


# Chi-Square Test for Independence

## Background
Mizzare Corporation has collected data on customer satisfaction levels for two types of smart home devices: Smart Thermostats and Smart Lights. We want to determine if there's a significant association between the type of device purchased and the customer's satisfaction level.

## Data Provided
The data is summarized in the following contingency table:

| Satisfaction       | Smart Thermostat | Smart Light | Total |
|--------------------|------------------|-------------|-------|
| Very Satisfied     | 50               | 70          | 120   |
| Satisfied          | 80               | 100         | 180   |
| Neutral            | 60               | 90          | 150   |
| Unsatisfied        | 30               | 50          | 80    |
| Very Unsatisfied   | 20               | 50          | 70    |
| **Total**          | 240              | 360         | 600   |

## Objective
To use the Chi-Square test for independence to determine if there's a significant association between the type of smart home device purchased (Smart Thermostats vs. Smart Lights) and the customer satisfaction level.

## Assignment Tasks

### 1. State the Hypotheses

- **Null Hypothesis (H0):** There is no association between the type of device purchased and the customer satisfaction level. The variables are independent.
- **Alternative Hypothesis (H1):** There is an association between the type of device purchased and the customer satisfaction level. The variables are dependent.

### 2. Compute the Chi-Square Statistic

We will calculate the Chi-Square statistic using the formula:
\[
\chi^2 = \sum \frac{(O_i - E_i)^2}{E_i}
\]
where \( O_i \) is the observed frequency and \( E_i \) is the expected frequency.

### 3. Determine the Critical Value

Using the significance level (α) of 0.05 and the degrees of freedom, which is calculated as:
\[
\text{Degrees of Freedom} = (r - 1) \times (c - 1)
\]
where \( r \) is the number of rows and \( c \) is the number of columns.

### 4. Make a Decision

Compare the Chi-Square statistic with the critical value to decide whether to reject the null hypothesis.

---

Below is the Python code that performs these steps and provides the result of the Chi-Square test.


In [5]:
import numpy as np
import pandas as pd
import scipy.stats as stats

# Data provided
data = np.array([
    [50, 70],
    [80, 100],
    [60, 90],
    [30, 50],
    [20, 50]
])

# Create a DataFrame for better readability
df = pd.DataFrame(data, columns=['Smart Thermostat', 'Smart Light'])
df.index = ['Very Satisfied', 'Satisfied', 'Neutral', 'Unsatisfied', 'Very Unsatisfied']

# Print the contingency table
print("Contingency Table:")
print(df)

# Calculate the Chi-Square statistic, p-value, degrees of freedom, and expected frequencies
chi2_statistic, p_value, dof, expected = stats.chi2_contingency(data)

# Determine the critical value for α = 0.05
alpha = 0.05
critical_value = stats.chi2.ppf(1 - alpha, df=dof)

# Print the results
print(f"\nChi-Square Statistic: {chi2_statistic:.4f}")
print(f"P-Value: {p_value:.4f}")
print(f"Degrees of Freedom: {dof}")
print(f"Expected Frequencies Table:\n{expected}")
print(f"Critical Value: {critical_value:.4f}")

# Make a decision
if chi2_statistic > critical_value:
    decision = "Reject the null hypothesis. There is a significant association between the type of device purchased and the customer satisfaction level."
else:
    decision = "Fail to reject the null hypothesis. There is not enough evidence to suggest an association between the type of device purchased and the customer satisfaction level."

print(f"\nDecision: {decision}")


Contingency Table:
                  Smart Thermostat  Smart Light
Very Satisfied                  50           70
Satisfied                       80          100
Neutral                         60           90
Unsatisfied                     30           50
Very Unsatisfied                20           50

Chi-Square Statistic: 5.6382
P-Value: 0.2278
Degrees of Freedom: 4
Expected Frequencies Table:
[[ 48.  72.]
 [ 72. 108.]
 [ 60.  90.]
 [ 32.  48.]
 [ 28.  42.]]
Critical Value: 9.4877

Decision: Fail to reject the null hypothesis. There is not enough evidence to suggest an association between the type of device purchased and the customer satisfaction level.


## Conclusion of Chi-Square Test for Independence

### Contingency Table

| Satisfaction       | Smart Thermostat | Smart Light |
|--------------------|------------------|-------------|
| Very Satisfied     | 50               | 70          |
| Satisfied          | 80               | 100         |
| Neutral            | 60               | 90          |
| Unsatisfied        | 30               | 50          |
| Very Unsatisfied   | 20               | 50          |

### Test Results

- **Chi-Square Statistic:** 5.6382
- **P-Value:** 0.2278
- **Degrees of Freedom:** 4
- **Expected Frequencies Table:**
- 
[[48. 72.]
[72. 108.]
[60. 90.]
[32. 48.]
[28. 42.]]

- **Critical Value (for α = 0.05):** 9.4877

### Decision

Since the Chi-Square statistic (5.6382) is less than the critical value (9.4877), we **fail to reject the null hypothesis**.

### Conclusion

Based on the results of the Chi-Square test, there is not enough evidence to suggest a significant association between the type of device purchased and the customer satisfaction level. This implies that the customer satisfaction levels are independent of the type of smart home device purchased, according to the provided data.
